frontendservice/src/views/CreateShop.vue

```diff
@@ -0,0 +1,166 @@
+          <button class="button is-link" type="submit" value="submit">Submit</button>
+        </div>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+import shopOwnerNavbar from "../components/NavBar-ShopOwner";
+import Vue from "vue";
+import VueTelInput from "vue-tel-input";
+import firebase from "firebase";
+
+Vue.use(VueTelInput);
+
+export default {
+  name: "CreateShop",
+  components: {
+    shopOwnerNavbar
+  },
+  data() {
+    return {
+      hidden: true,
+      isCorrect: false,
+      phoneNumber: 0
+    };
+  },
+  methods: {
+    createShop: function() {
+      var shopName = document.getElementById("shopName").value;
+      var address = document.getElementById("address").value;
+      var shopType = document.getElementById("shopType").value;
+      var checkBox = document.getElementById("conditions-check");
+      var phoneNo = this.phoneNumber;
+      console.log(document.getElementById("phoneNumber"));
+      // Regex check only permit # . characters, digits and space
+      // eg - Haridwar123 is valid.
+      var addressRegex = new RegExp(/^[#.a-zA-Z0-9 ]+$/i);
+      // Regex check only permit characters, digits and space
+      // eg - Ishita Sweets is valid.
+      var nameRegex = new RegExp(/^[a-zA-Z0-9 ]+$/i);
+      var isError = false;
+
+      if (!checkBox.checked) {
+        var checkBoxError = document.createElement("div");
+        checkBoxError.innerHTML = "Please check this to continue.";
+        document.getElementById("msg").appendChild(checkBoxError);
+        isError = true;
+      }
+      if (!addressRegex.test(address)) {
+        var addressError = document.createElement("div");
+        addressError.innerHTML = "Invalid address.";
+        document.getElementById("msg").appendChild(addressError);
+        isError = true;
+      }
+      if (!nameRegex.test(shopName)) {
+        var nameError = document.createElement("div");
+        nameError.innerHTML = "Invalid Shop Name.";
+        document.getElementById("msg").appendChild(nameError);
+        isError = true;
+      }
+      if (!this.isCorrect) {
+        var phoneNumberError = document.createElement("div");
+        phoneNumberError.innerHTML = "Invalid Phone Number.";
+        document.getElementById("msg").appendChild(phoneNumberError);
+        isError = true;
+      }
+      if (isError) {
+        return;
+      }
+      var self = this;
+      axios
+        .post("http://penguin.termina.linux.test:8080/shop", {
+          shopOwnerId: firebase.auth().currentUser.uid,
+          shopName: shopName,
+          address: address,
+          phoneNumber: phoneNo,
+          shopType: shopType
+        })
+        .then(function(response) {
+          // TODO: Exception handling.
+          // TODO: Clear phone number.
+          document.getElementById("msg").innerHTML = "Shop Added!";
+          document.getElementById("shopName").value = "";
+          document.getElementById("address").value = "";
```

Comment:
"What about clearing phone number?" - This was the comment that Sugandha had added. I'm not sure if you have implemented this. If done, you can resolve this.

Reply:
Actually I have added this as a TODO for implementing later as this may take a while. Have discussed this with Sugandha.

Reply:
Oh sure.